File: PenetrationProjectAPI/src/main/java/com/ssafy/cafe/model/service/ShoppingListService.java
package com.ssafy.cafe.model.service;

import java.util.List;
import java.util.Map;

import com.ssafy.cafe.model.dto.ShoppingList;

public interface ShoppingListService {

	public List<Map<String, Object>> add(Map<String, Object> map);

	public List<Map<String, Object>> update(ShoppingList shoppingList);

	public List<Map<String, Object>> removeByUser(String userId);

	public List<Map<String, Object>> removeOne(Map<String, Object> map);

	public List<Map<String, Object>> selectByUser(String userId);

}

File: app/src/main/java/com/ssafy/smartstore/data/remote/api/ShoppingListService.kt
package com.ssafy.smartstore.data.remote.api

import com.ssafy.smartstore.model.OrderProduct
import com.ssafy.smartstore.util.Network
import retrofit2.Response
import retrofit2.http.*

interface ShoppingListService {

    @POST("${Network.SHOPPING_LIST}/add")
    suspend fun addShoppingList(@Body map: HashMap<String, Any>): Response<List<OrderProduct>>

    @GET("${Network.SHOPPING_LIST}/{userId}")
    suspend fun selectByUser(@Path("userId") userId: String): Response<List<OrderProduct>>

    @POST("${Network.SHOPPING_LIST}/delete")
    suspend fun deleteOneItem(@Body map: HashMap<String, Any>): Response<List<OrderProduct>>

    @DELETE("${Network.SHOPPING_LIST}/{userId}")
    suspend fun deleteByUser(@Path("userId") userId: String): Response<List<OrderProduct>>

}